src/overlays/actors/ovl_Bg_Mori_Bigst/z_bg_mori_bigst.c

```diff
@@ -20,39 +36,219 @@ const ActorInit Bg_Mori_Bigst_InitVars = {
+}
+
+void BgMoriBigst_Fall(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Actor_MoveForward(&this->dyna.actor);
+    if (this->dyna.actor.posRot.pos.y <= this->dyna.actor.initPosRot.pos.y) {
+        this->dyna.actor.posRot.pos.y = this->dyna.actor.initPosRot.pos.y;
+        BgMoriBigst_Quake(this, globalCtx);
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_STONE_BOUND);
+        func_800800F8(globalCtx, 0x3FC, 8, &this->dyna.actor, 0);
+        func_8002DF38(globalCtx, NULL, 0x3C);
+    }
+}
+
+void BgMoriBigst_Quake(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Camera** cameras = globalCtx->cameraPtrs;
+    s32 quake;
+
+    BgMoriBigst_SetActionFunction(this, BgMoriBigst_SetupSpawnStalfosPair);
+    this->waitTimer = 18;
+    quake = Quake_Add(cameras[globalCtx->activeCamera], 3);
+    Quake_SetSpeed(quake, 0x61A8);
+    Quake_SetQuakeValues(quake, 5, 0, 0, 0);
+    Quake_SetCountdown(quake, 16);
+}
+
+void BgMoriBigst_SetupSpawnStalfosPair(BgMoriBigst* this, GlobalContext* globalCtx) {
+    if (this->waitTimer <= 0) {
+        BgMoriBigst_SpawnStalfosPair(this, globalCtx);
+    }
+}
+
+void BgMoriBigst_SpawnStalfosPair(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Actor* stalfos1;
+    Actor* stalfos2;
+
+    BgMoriBigst_SetActionFunction(this, BgMoriBigst_StalfosPairWait);
+    Flags_UnsetClear(globalCtx, this->dyna.actor.room);
+    stalfos1 =
+        Actor_SpawnAsChild(&globalCtx->actorCtx, &this->dyna.actor, globalCtx, 2, 70.0f, 827.0f, -3383.0f, 0, 0, 0, 5);
+    if (stalfos1 != NULL) {
+        this->dyna.actor.child = NULL;
+        this->dyna.actor.initPosRot.rot.z++;
+    } else {
+        // Warning: 3-1 Stalfos failure
+        osSyncPrintf("Warning : 第３-1スタルフォス発生失敗\n");
+    }
+    stalfos2 =
+        Actor_SpawnAsChild(&globalCtx->actorCtx, &this->dyna.actor, globalCtx, 2, 170.0f, 827.0f, -3260.0f, 0, 0, 0, 5);
+    if (stalfos2 != NULL) {
+        this->dyna.actor.child = NULL;
+        this->dyna.actor.initPosRot.rot.z++;
+    } else {
+        // Warning: 3-2 Stalfos failure
+        osSyncPrintf("Warning : 第３-2スタルフォス発生失敗\n");
+    }
+    Flags_SetClear(globalCtx, this->dyna.actor.room);
+}
+
+void BgMoriBigst_StalfosPairWait(BgMoriBigst* this, GlobalContext* globalCtx) {
+    if ((this->dyna.actor.initPosRot.rot.z == 0) && !func_8008E988(globalCtx)) {
+        Flags_SetSwitch(globalCtx, (this->dyna.actor.params >> 8) & 0x3F);
+        BgMoriBigst_FinalNoop(this, globalCtx);
+    }
+}
+
+void BgMoriBigst_FinalNoop(BgMoriBigst* this, GlobalContext* globalCtx) {
+    BgMoriBigst_SetActionFunction(this, NULL);
+}
+
+void BgMoriBigst_Update(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    BgMoriBigst* this = THIS;
+
+    Actor_SetHeight(&this->dyna.actor, 50.0f);
+    if (this->waitTimer > 0) {
+        this->waitTimer--;
+    }
+    if (func_80043590(&this->dyna)) {
+        func_80074CE8(globalCtx2, 6);
+    }
```

Comment:
Just a thought, this may be "if player walking on me",
`func_80043590` is `dynaActor->unk_160 & 4 ? 1 : 0`
`func_800434F8` is `dynaActor->unk_160 |= 4` and is only called by `func_80043508`:
```c
void func_80043508(CollisionContext* colCtx, u32 floorPolySource) {
    DynaPolyActor* dynaActor = DynaPolyInfo_GetActor(colCtx, floorPolySource);

    if (dynaActor != NULL) {
        func_800434F8(dynaActor);
    }
}
```
which is only called from player, apparently [like this](https://github.com/Roman971/oot/blob/player/src/overlays/actors/ovl_player_actor/z_player.c#L9471):
`func_80043508(&globalCtx->colCtx, this->actor.floorPolySource);`

flags are a mess lol

Reply:
what does `func_80074CE8` do?

Reply:
no clue, envCtx is FeelsUnkMan

https://github.com/jacobly0/oot/blob/kankyo/src/code/z_kankyo.c#L1220